Source/WebKit/NetworkProcess/cache/NetworkCache.cpp

```diff
@@ -284,6 +284,12 @@ static StoreDecision makeStoreDecision(const WebCore::ResourceRequest& originalR
             return StoreDecision::NoDueToHTTPStatusCode;
     }
 
+    // Our caching does not take into account fragments properly.
+    // If we cache localhost/#key=foo we will return the same parameters key=foo
+    // on the next visit to the domain even if the parameters have been updated.
+    if (response.isRedirection() && originalRequest.url().hasFragmentIdentifier())
+        return StoreDecision::NoDueToRedirectContainingFragments;
```

Comment:
The redirect does not necessarily contain a fragment, it is the original request that has a fragment.